datadog_lambda/tracing.py

```diff
@@ -129,16 +130,27 @@ def extract_context_from_lambda_context(lambda_context):
     dd_trace libraries inject this trace context on synchronous invocations
     """
     client_context = lambda_context.client_context
-
+    trace_id = None  # type: Optional[str]
+    parent_id = None  # type: Optional[str]
+    sampling_priority = None  # type: Optional[str]
```

Comment:
I think the "str" type is a bit complicated, because it should be 'unicode' in python 2.7, and 'str' in 3+. It seems like the correct type would be 'Text' https://mypy.readthedocs.io/en/stable/cheat_sheet.html#built-in-types

Not sure it's worth introducing type hinting in this PR. I think it would make sense to start at the same time we drop 2.7 support, so we can use the more convenient 3+ syntax

Reply:
That makes sense. My instinct is to scatter type hints in as many places as possible, but it's probably better to wait until the 2 -> 3 conversion.